parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -60,33 +66,40 @@ std::optional<range> get_range(const context::hlasm_statement* stmt,
     return std::nullopt;
 }
 
-size_t& emplace_hit_count(
-    hit_count_map& hc_map, const utils::resource::resource_location& rl, range r, bool can_be_external_macro)
+void emplace_hit_count(hit_count_map& hc_map,
+    const utils::resource::resource_location& rl,
+    stmt_lines_range lines_range,
+    bool can_be_external_macro,
+    bool increase_hit_count)
 {
-    static constexpr auto map_emplacer = []<typename VALUE>(auto& m, const auto& k) {
-        if (auto m_it = m.find(k); m_it != m.end())
-            return m_it;
+    auto& start_line = lines_range.first;
+    auto& end_line = lines_range.second;
 
-        return m.try_emplace(k, VALUE {}).first;
-    };
+    auto& stmt_hc_details = hc_map.try_emplace(rl).first->second;
+    stmt_hc_details.is_external_macro &= can_be_external_macro;
+    stmt_hc_details.max_lineno = std::max(stmt_hc_details.max_lineno, start_line);
 
-    auto& stmt_hc_details = map_emplacer.template operator()<stmt_hit_count_details>(hc_map, rl)->second;
-    auto& [stmt_r, stmt_count] =
-        map_emplacer.template operator()<hit_count_details>(stmt_hc_details.stmt_hc_map, r.start.line)->second;
+    auto& line_hits = stmt_hc_details.line_hits;
+    while (line_hits.size() <= end_line)
+        line_hits.resize(2 * line_hits.size());
 
-    stmt_hc_details.is_external_macro &= can_be_external_macro;
-    stmt_r = std::move(r);
-    return stmt_count;
+    std::for_each_n(
+        std::next(line_hits.begin(), start_line), end_line - start_line + 1, [increase_hit_count](auto& line_hit) {
+            line_hit.contains_statement = true;
+
+            if (increase_hit_count)
+                line_hit.count++;
+        });
```

Comment:
why skip the start line?
Also simple way is sometimes the way :)
```suggestion
    for (auto i = start_line; i <= end_line; ++i)
    {
        auto& line_hit = line_hits[i];
        line_hit.contains_statement = true;
        if (increase_hit_count)
            line_hit.count++;
    }
```